### PR TITLE
mgr: fix the problem that slow op warning does not display correctly

### PR DESCRIPTION
**Pull request by IlsooByun**
Fixes: https://tracker.ceph.com/issues/41741
Signed-off-by: Ilsoo Byun <ilsoobyun@linecorp.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

File: src/mgr/DaemonHealthMetricCollector.h [hunk 1]
```diff
@@ -12,7 +12,7 @@ class DaemonHealthMetricCollector {
   static std::unique_ptr<DaemonHealthMetricCollector> create(daemon_metric m);
   void update(const DaemonKey& daemon, const DaemonHealthMetric& metric) {
     if (_is_relevant(metric.get_type())) {
-      reported = _update(daemon, metric);
+      if (_update(daemon, metric)) reported = true;
```
Comment (sebastian-philipp) [inline src/mgr/DaemonHealthMetricCollector.h]: ar, can you put that into a new line?

**Comment by sebastian-philipp** [comment]
jenkins retest this please

File: src/mgr/DaemonHealthMetricCollector.h (hunk 1 above)
Comment (IlsooByun) [reply src/mgr/DaemonHealthMetricCollector.h]: I pushed it again but I don't know if I understood you correctly.

File: src/mgr/DaemonHealthMetricCollector.h [hunk 2]
```diff
@@ -12,7 +12,8 @@ class DaemonHealthMetricCollector {
   static std::unique_ptr<DaemonHealthMetricCollector> create(daemon_metric m);
   void update(const DaemonKey& daemon, const DaemonHealthMetric& metric) {
     if (_is_relevant(metric.get_type())) {
-      reported = _update(daemon, metric);
+      if (_update(daemon, metric)) 
+        reported = true;
```
Comment (tchaikov) [inline src/mgr/DaemonHealthMetricCollector.h]: how about 
```c++
reported |= _update(daemon, metric);
```
?

**Comment by tchaikov** [comment]
could you prefix the title of your commit message with the subcomponent your are changing ? see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes, in this case, it would be "mgr:".

and the title of the commit message could be improved. i'd recommend

> do not reset `reported` if a new metric is not collected

**Comment by IlsooByun** [comment]
> could you prefix the title of your commit message with the subcomponent your are changing ? see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes, in this case, it would be "mgr:".
> 
> and the title of the commit message could be improved. i'd recommend
> 
> > do not reset `reported` if a new metric is not collected

I modified it. Thank you for letting me know.

File: src/mgr/DaemonHealthMetricCollector.h (hunk 2 above)
Comment (IlsooByun) [reply src/mgr/DaemonHealthMetricCollector.h]: I fixed it as you said.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-09-12_02:56:40-rados-wip-kefu-testing-2019-09-11-2224-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-09-13_04:19:52-rados-wip-kefu-testing-2019-09-11-2224-distro-basic-mira/

i don't think the failures are related.

**Pull request merged**